flocker/acceptance/test_restarts.py

```diff
@@ -0,0 +1,169 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for restarting and reboots and their interactions.
+"""
+
+import os
+
+from subprocess import call, check_call, CalledProcessError
+from unittest import SkipTest
+
+from twisted.python.filepath import FilePath
+from twisted.trial.unittest import TestCase, FailTest
+
+from treq import get, content
+
+from ..testtools import loop_until, random_name
+from .testtools import create_dataset, require_cluster
+from ..common.runner import run_ssh
+
+
+REBOOT_SERVER = FilePath(__file__).sibling(b"reboot_httpserver.py")
+
+def _service(address, name, action):
+    """
+    :param bytes address: The public IP of the node on which the service will
+        be changed.
+    :param bytes action: The action to perform on the service.
+    """
+    from twisted.internet import reactor
+    command = ["systemctl", action, name]
```

Comment:
Add an Ubuntu compatible version of this. 
Figure out how to disable / enable services on Ubuntu.